tool.go

```diff
@@ -772,6 +796,25 @@ func printError(err error, options *gbuild.Options, browserErrors *bytes.Buffer)
 	}
 }
 
+// verifyAndUnsetGOARCH verifies that GOARCH environment value is not set to
+// an unsupported value. It also unsets it, if set, because the gopherjs compiler
+// expects build.Default.GOARCH to actually be something like amd64 rather than js.
+func verifyAndUnsetGOARCH() error {
+	goarch, ok := os.LookupEnv("GOARCH")
+	if !ok {
+		return nil
+	}
+	if goarch != "js" {
+		return fmt.Errorf("gopherjs: unsupported GOOS/GOARCH pair %s/%s", build.Default.GOOS, goarch)
+	}
+	err := os.Unsetenv("GOARCH")
+	if err != nil {
+		return err
+	}
+	build.Default.GOARCH = runtime.GOARCH // Reset GOARCH to its default value, what it would've been had the GOARCH environment variable not overridden it.
```

Comment:
I'll investigate, thanks for the improvement idea.

I worry that some packages in std lib might list all known architectures explicitly, and `js` wouldn't be one of them, so we would select no .go files, unless we pick a well known architecture such as `amd64`. But if it's only a few such packages, we can get around by making them special cases.

I'll see if it works out.